### DISCUSSION TITLE
disable the option to display AP Mode from the captive portal

### DISCUSSION THREAD

**Discussion by microfoundry** [mathieucarbou/MycilaESPConnect#32] [Ideas]
Hey Mathieu - would you consider 2 options:

1. Defines for the preference keys? I'd like to be able to utilize preferences keys I've established in my current configuration...
2. A marco to disable the option to display AP Mode from the captive portal? In my case I need the user to set the device as a WiFi station and I'd like to reduce the opportunity for human error...

-Terry

**Comment by mathieucarbou** [comment]
> 1. Defines for the preference keys? I'd like to be able to utilize preferences keys I've established in my current configuration...

This is supported: look at the documentation in the header and the example. That is also what I use in my apps ;-)

> A marco to disable the option to display AP Mode from the captive portal

Currently this is not possible to change the UI depending on some config parameters.
But if you want to refactor the library to do the work, I will be glad to review.

Also note that you don't need to disable it: if the user is choosing AP mode, in the callback, you can always override that and re-activate the captive portal. You can be in a complete control over the parameters you decide to save and then load and pass to the lib.

You should maybe play a bit more with it and the examples and see what it can do....
